### PR TITLE
Llu/smem elog fail 4090

### PR DESCRIPTION
**Pull request by liqiangxl**
In test cases, change `use_smem_epilogue` from `true` to set by `generateSharedMemoryEpilogueHeuristics` with `ignore_occupancy_drop = true`. This allows us to test the use_smem_epilogue implementation if the device can launch at least 1 block per SM. Otherwise will test with `use_smem_epilogue = false` and mark the case as `SKIPPED`.

Tested on 3080 @ nvdl-bsg-amd02
[  SKIPPED ] 3 tests, listed below:
[  SKIPPED ] NVFuserTest.FusionAmpereMatmulSmemEpilogueCast_CUDA
[  SKIPPED ] NVFuserTest.FusionAmpereMatmulSmemEpilogueRelu_CUDA
[  SKIPPED ] MatmulSASSTest.AmpereModifiersSharedMemoryEpilogue_CUDA

### CODE REVIEW

File: csrc/scheduler/mma_utils.cpp [hunk 1]
```diff
@@ -22,17 +22,18 @@ namespace mma_utils {
 bool generateSharedMemoryEpilogueHeuristics(
     const MatMulTileOptions& gemm_tile,
     const int smem_double_buffer_stage,
-    const MmaDataTypes& data_types) {
+    const MmaDataTypes& data_types,
+    const bool ignore_occupancy_drop) {
   const auto properties = at::cuda::getCurrentDeviceProperties();
   const size_t device_smem_limit = properties->sharedMemPerBlockOptin;
+  // 1024 bytes are reserved for kernel launch overhead
+  const size_t shared_memory_overhead = 1024;
```
Comment (zasdfgbnm) [inline csrc/scheduler/mma_utils.cpp:30]: Should we use this?
https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaDeviceProp.html#structcudaDeviceProp_15a1f6049e9e9fd8af0d2a8a5df289ab7

Comment (liqiangxl) [reply csrc/scheduler/mma_utils.cpp:30]: yes!

File: test/test_matmul_sass.cpp [hunk 2]
```diff
@@ -206,121 +209,251 @@ TEST_F(MatmulSASSTest, AmpereModifiers_CUDA) {
...
-                      i.modifiers() == expect,
-                      "Modifiers for DEPBAR has changed. "
-                      "Please manually check if the new modifiers makes sense and update this test. "
-                      "Expect: ",
-                      expect,
-                      " Get: ",
-                      i.modifiers());
-                  found_DEPBAR = true;
-                }
+  bool found_LDGSTS = false;
```
Comment (liqiangxl) [inline test/test_matmul_sass.cpp:212]: Reversed the change to `AmpereModifiers_CUDA` and seperate the use_smem_epilogue test to a new test.

**Review by zasdfgbnm — APPROVED** [review]
Thanks for fixing!

**Pull request merged**